### ISSUE TITLE
[core][release] dask_on_ray_1tb/100gb_sort failing with WorkerCrashedError on OOM

### ISSUE THREAD

**Issue by rickyyx** [ray-project/ray#28781]
### What happened + What you expected to happen

[dask_on_ray_1tb_sort](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_ANmBVP1kpGqqATfPzApXgurw?command-history-section=command_history)
[dask_on_ray_100gb_sort](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_HqitSduBRqAf5d588txj9x2F?command-history-section=command_history)

```
(raylet, ip=10.0.3.179) [2022-09-22 17:46:53,692 E 134 134] (raylet) node_manager.cc:2966: System memory low at node with IP 10.0.3.179. Used memory (216.78GB) / total capacity (219.60GB) (0.98715) exceeds threshold 0.95, killing latest task with name dask:('head-partial-10-sort_index-be01da623ff06afe98fde4f50eda005f', 586) and task ID NIL_ID to avoid running out of memory.
(raylet, ip=10.0.3.179) This may indicate a memory leak in a task or actor, or that too many tasks are running in parallel.
(raylet, ip=10.0.3.179) To find the highest memory consumers, use `ray logs raylet.out -ip 10.0.3.179`.
(raylet, ip=10.0.3.179) Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the eviction threshold, set the environment variable `RAY_memory_usage_threshold_fraction` when starting Ray. To disable worker eviction, set the environment variable `RAY_memory_monitor_interval_ms` to zero.
(raylet, ip=10.0.3.179) [2022-09-22 17:47:04,585 E 134 134] (raylet) node_manager.cc:2966: System memory low at node with IP 10.0.3.179. Used memory (218.38GB) / total capacity (219.60GB) (0.994424) exceeds threshold 0.95, killing latest task with name dask:('sort_index-be01da623ff06afe98fde4f50eda005f', 479) and task ID ef70f211e3dbf2f451cc5079e1daed7e4281a4de02000000 to avoid running out of memory.
(raylet, ip=10.0.3.179) This may indicate a memory leak in a task or actor, or that too many tasks are running in parallel.
(raylet, ip=10.0.3.179) To find the highest memory consumers, use `ray logs raylet.out -ip 10.0.3.179`.
(raylet, ip=10.0.3.179) Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the eviction threshold, set the environment variable `RAY_memory_usage_threshold_fraction` when starting Ray. To disable worker eviction, set the environment variable `RAY_memory_monitor_interval_ms` to zero.
(raylet, ip=10.0.3.179) [2022-09-22 17:47:14,868 E 134 134] (raylet) local_object_manager.cc:362: Failed to send object spilling request: GrpcUnavailable: RPC Error message: Socket closed; RPC Error details: 
....
Traceback (most recent call last):
  File "dask_on_ray/dask_on_ray_sort.py", line 230, in <module>
    file_path=args.file_path,
  File "dask_on_ray/dask_on_ray_sort.py", line 145, in trial
    10, npartitions=-1
  File "/home/ray/anaconda3/lib/python3.7/site-packages/dask/dataframe/core.py", line 1140, in head
    return self._head(n=n, npartitions=npartitions, compute=compute, safe=safe)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/dask/dataframe/core.py", line 1174, in _head
    result = result.compute()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/dask/base.py", line 290, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/dask/base.py", line 573, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/dask/scheduler.py", line 203, in ray_dask_get
    result = ray_get_unpack(object_refs, progress_bar_actor=pb_actor)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/dask/scheduler.py", line 504, in ray_get_unpack
    computed_result = get_result(object_refs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/dask/scheduler.py", line 491, in get_result
    return ray.get(object_refs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 2279, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError: ray::dask:('head-1000-10-sort_index-be01da623ff06afe98fde4f50eda005f', 0) (pid=1121, ip=10.0.3.15)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RayTaskError: ray::dask:('head-partial-10-sort_index-be01da623ff06afe98fde4f50eda005f', 746) (pid=1121, ip=10.0.3.15)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RayTaskError: ray::dask:('sort_index-be01da623ff06afe98fde4f50eda005f', 746) (pid=1121, ip=10.0.3.15)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.WorkerCrashedError: The worker died unexpectedly while executing this task. Check python-core-worker-*.log files for more information.
```

### Versions / Dependencies

Last success: [45d7cd2](https://github.com/ray-project/ray/commit/45d7cd2f86d0eccdad632ac04514279f4d126106)
First failure: [fb7472f](https://github.com/ray-project/ray/commit/fb7472f047d669ec964bcb66f5f718c68609864b)

fb7472f04 Remove RAY_RAYLET_NODE_ID (#28715)
93f911eb6 Add API latency and call counts metrics to dashboard APIs (#28279)
66aae4cb3 [Release Test] Make sure to delete all EBS volumes (#28707)
697df80a0 [Serve] [Docs] Remove incorrect output (#28708)
d8c9aa777 [docs] configurable ecosystem gallery (#28662)
42874e1fc [RLlib] Atari gym environments now require ale-py. (#28703)
b7f0346d5 [AIR] Maintain dtype info in LightGBMPredictor (#28673)
f6ae7ee44 [tune] Test background syncer serialization (#28699)
87f22e106 [ci] Requirements contains duplicate of 'starlette' (#28698)
2e7040ef1 [Tune] [PBT] Maintain consistent `Trial`/`TrialRunner` state when pausing and resuming trial (#28511)
653063555 [ci] Fix mac pipeline (use python 2 in CI scripts) (#28695)
ee2a8da87 [ci] Move to new hierarchical docker structure + pipeline (#28641)
a3c97b44e [Doc] Revamp ray core design patterns doc [8/n]: pass large arg by value (#28660)
db2ce69e4 [Datasets] Add initial aggregate benchmark (#28486)
9c2abf916 [KubeRay][Operator] Improve migration notes (#28672)
45d7cd2f8 [core] Support generators to allow tasks to return a dynamic number of objects (#28291)

### Reproduction script

NA
